shenyu-plugin/shenyu-plugin-rpc-response/src/main/java/org/apache/shenyu/plugin/rpc/response/RpcResponsePlugin.java

```diff
@@ -0,0 +1,113 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shenyu.plugin.rpc.response;
+
+import org.apache.shenyu.common.constant.Constants;
+import org.apache.shenyu.common.enums.PluginEnum;
+import org.apache.shenyu.common.enums.RpcTypeEnum;
+import org.apache.shenyu.common.utils.JsonUtils;
+import org.apache.shenyu.plugin.api.ShenyuPlugin;
+import org.apache.shenyu.plugin.api.ShenyuPluginChain;
+import org.apache.shenyu.plugin.api.context.ShenyuContext;
+import org.apache.shenyu.plugin.api.result.ShenyuResultEnum;
+import org.apache.shenyu.plugin.api.result.ShenyuResultWrap;
+import org.apache.shenyu.plugin.api.utils.WebFluxResultUtils;
+import org.springframework.web.server.ServerWebExchange;
+import reactor.core.publisher.Mono;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * this is rpc response plugin.
+ */
+public class RpcResponsePlugin implements ShenyuPlugin {
+
+    private static final Map<String, String> RPC_TYPE_MAP = new HashMap<String, String>() {
+        {
+            put(RpcTypeEnum.GRPC.getName(), Constants.GRPC_RPC_RESULT);
+            put(RpcTypeEnum.SOFA.getName(), Constants.SOFA_RPC_RESULT);
+            put(RpcTypeEnum.DUBBO.getName(), Constants.DUBBO_RPC_RESULT);
+            put(RpcTypeEnum.TARS.getName(), Constants.TARS_RPC_RESULT);
+        }
+    };
+
+    /**
+     * Process the Web request and (optionally) delegate to the next
+     * {@code WebFilter} through the given {@link ShenyuPluginChain}.
+     *
+     * @param exchange the current server exchange
+     * @param chain    provides a way to delegate to the next filter
+     * @return {@code Mono<Void>} to indicate when request processing is complete
+     */
+    @Override
+    public Mono<Void> execute(final ServerWebExchange exchange, final ShenyuPluginChain chain) {
+        return chain.execute(exchange).then(Mono.defer(() -> {
+            final Object result = Optional.ofNullable(getResult(exchange)).map(data -> exchange.getAttribute(data)).orElse(null);
+            if (Objects.isNull(result)) {
+                Object error = ShenyuResultWrap.error(ShenyuResultEnum.SERVICE_RESULT_ERROR.getCode(), ShenyuResultEnum.SERVICE_RESULT_ERROR.getMsg(), null);
+                return WebFluxResultUtils.result(exchange, error);
+            }
+            Object success = ShenyuResultWrap.success(ShenyuResultEnum.SUCCESS.getCode(), ShenyuResultEnum.SUCCESS.getMsg(), JsonUtils.removeClass(result));
+            return WebFluxResultUtils.result(exchange, success);
+        }));
+    }
+
+    private String getResult(final ServerWebExchange exchange) {
+        ShenyuContext shenyuContext = exchange.getAttribute(Constants.CONTEXT);
+        return Optional.ofNullable(shenyuContext).map(data -> RPC_TYPE_MAP.get(data.getRpcType())).orElse(null);
```

Comment:
I am not sure that if every roc plugin use this key to store the response, I am afraid that it would be a little weird. And if u change here, other plugin's response should store in this key too.

Reply:
This logic works the same way as the previous four plug-ins